source/common/http/header_utility.h

```diff
@@ -198,13 +198,20 @@ class HeaderUtility {
    */
   static absl::string_view::size_type getPortStart(absl::string_view host);
 
-  /* Does a common header check ensuring required headers are present.
+  /* Does a common header check ensuring required request headers are present.
    * Required request headers include :method header, :path for non-CONNECT requests, and
    * host/authority for HTTP/1.1 or CONNECT requests.
    * @return Status containing the result. If failed, message includes details on which header was
    * missing.
    */
-  static Http::Status checkRequiredHeaders(const Http::RequestHeaderMap& headers);
+  static Http::Status checkRequiredRequestHeaders(const Http::RequestHeaderMap& headers);
```

Comment:
I hear your point about calling this from the codec being really messy.
optional!: Could we at least as part of this PR make the codecs not throw/crash if they get bad data a well?  I think we'd prefer to serialize 0 if status code is not present to crashing though Matt may disagree =P

Reply:
In order to do so, we should eliminate all the usage of `getResponseStatus` and use `getResponseStatusNoThrow` instead? e.g.
- https://github.com/envoyproxy/envoy/blob/5c51bf16660db4713f8879faf0d34e2075739273/source/common/http/http2/codec_impl.cc#L336
- https://github.com/envoyproxy/envoy/blob/5c51bf16660db4713f8879faf0d34e2075739273/source/common/http/http1/codec_impl.cc#L373-L375

I think fixing this is worth a separate PR and discussion, so can I open an issue for now and discuss there?

Reply:
SGTM, and as it's optional don't feel obliged to tackle it (though it would be nice!)
I think it's largely a part of removing exceptions from the data plane, which I think @chaoqin-li1123 is looking at